controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -432,6 +432,35 @@ public void getDelegationToken(StreamInfo request, StreamObserver<DelegationToke
                 responseObserver);
     }
 
+    // region watermarking apis
+    
+    @Override
+    public void noteTimestampFromWriter(Controller.TimestampFromWriter request, StreamObserver<Controller.TimestampResponse> responseObserver) {
+        StreamInfo streamInfo = request.getPosition().getStreamInfo();
+        log.info("noteWriterMark called for stream {}/{}, writer={} time={}", streamInfo.getScope(),
+                streamInfo.getStream(), request.getWriter(), request.getTimestamp());
+        authenticateExecuteAndProcessResults(() -> this.authHelper.checkAuthorization(
```

Comment:
CheckAuthorizationAndCreateToken is used when we want to send token back to client to enable them to connect to segment store nodes. 
So it is relevant for `getCurrentSegments`, `getSuccessor` like apis. For apis where client simply asks controller to perform some action on its behalf, it could be a get or an update request, there is no need to generate token.